src/c2goto/library/io.c

```diff
@@ -61,8 +71,19 @@ __ESBMC_HIDE:;
 char *fgets(char *str, int size, FILE *stream)
 {
 __ESBMC_HIDE:;
-  _Bool error;
-  return error ? 0 : str;
+
+  //do nothing, report error
+  if(size < 2)
+    return NULL;
+
+  // add nondeterministic characters to str
+  for(int i = 0; i <= (size - 1); i++)
```

Comment:
Should also stop after a new-line character is encountered (and on EOF).

Reply:
Fixed.